homeassistant/components/azure_data_explorer/__init__.py

```diff
@@ -0,0 +1,215 @@
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.util.dt import utcnow
+
+from .client import AzureDataExplorerClient
+from .const import (
+    CONF_FILTER,
+    CONF_SEND_INTERVAL,
+    DATA_FILTER,
+    DATA_HUB,
+    DEFAULT_MAX_DELAY,
+    DOMAIN,
+    FILTER_STATES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional(CONF_FILTER, default={}): FILTER_SCHEMA,
+            },
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, yaml_config: ConfigType) -> bool:
+    """Activate ADX component from yaml.
+
+    Adds an empty filter to hass data.
+    Tries to get a filter from yaml, if present set to hass data.
+    If config is empty after getting the filter, return, otherwise emit
+    deprecated warning and pass the rest to the config flow.
+    """
+
+    hass.data.setdefault(DOMAIN, {DATA_FILTER: FILTER_SCHEMA({})})
+    if DOMAIN not in yaml_config:
+        return True
+    hass.data[DOMAIN][DATA_FILTER] = yaml_config[DOMAIN][CONF_FILTER]
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Do the setup based on the config entry and the filter from yaml."""
+    hass.data.setdefault(DOMAIN, {DATA_FILTER: FILTER_SCHEMA({})})
+    adx = AzureDataExplorer(hass, entry)
+    try:
+        await adx.test_connection()
+    except KustoServiceError as exp:
+        raise ConfigEntryError(
+            "Could not find Azure Data Explorer database or table"
+        ) from exp
+    except KustoAuthenticationError:
+        return False
+
+    hass.data[DOMAIN][DATA_HUB] = adx
+    entry.async_on_unload(entry.add_update_listener(async_update_listener))
+    await adx.async_start()
+    return True
+
+
+async def async_update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Update listener for options."""
+    hass.data[DOMAIN][DATA_HUB].update_options(entry.options)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    adx = hass.data[DOMAIN].pop(DATA_HUB)
+    await adx.async_stop()
+    return True
+
+
+class AzureDataExplorer:
+    """A event handler class for Azure Data Explorer."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the listener."""
+
+        self.hass = hass
+        self._entry = entry
+        self._entities_filter = hass.data[DOMAIN][DATA_FILTER]
+
+        self._client = AzureDataExplorerClient(**entry.data)
+
+        self._send_interval = entry.options[CONF_SEND_INTERVAL]
+        self._max_delay = DEFAULT_MAX_DELAY
+
+        self._shutdown = False
+        self._queue: asyncio.PriorityQueue[
+            tuple[int, tuple[datetime, State | None]]
+        ] = asyncio.PriorityQueue()
```

Comment:
```suggestion
        self._queue: asyncio.Queue[State] = asyncio.Queue()
```

The priority is always the same when adding the state to the queue. Therefore, we can use a normal queue